lib/portlayer/storage/vsphere/datastore.go

```diff
@@ -232,35 +255,23 @@ func (d *datastore) mkRootDir(ctx context.Context, rootdir string) error {
 		}
 
 		// set the root url to the UUID of the dir we created
-		d.rootdir = path.Base(uuid)
-		d.rooturl = d.ds.Path(d.rootdir)
-		log.Infof("Created image store parent directory (%s) at %s", rootdir, d.rooturl)
-		return nil
-	}
-
-	// Handle regular local datastore
-	// check if it already exists
-	_, err := d.Ls(ctx, d.ds.Path(rootdir))
-	if err != nil && !types.IsFileNotFound(err) {
-		return err
-	}
-
-	d.rootdir = rootdir
-	d.rooturl = d.ds.Path(rootdir)
+		d.RootURL = d.ds.Path(path.Base(uuid))
```

Comment:
Am I correct in thinking this is not actually a URL given the rootDir implementation? Otherwise I'd expect it to be used directly as `d.RootURL.Path`:
```
func (d *Datastore) rootDir() string {
 	return strings.SplitN(d.RootURL, " ", 2)[1]
 }
```